draft-ietf-ppm-dap.md

```diff
@@ -400,6 +402,29 @@ to error types, rather than the full URNs. For example, an "error of type
 This document uses the verbs "abort" and "alert with `[some error message]`" to
 describe how protocol participants react to various error conditions.
 
+## HTTPS Sender Authentication {#https-sender-auth}
+
+Some HTTP requests in the PPM protocol require the sender to authenticate its
+request. It does so as described here.
+
+Prior to the start of the protocol, the sender and receiver arrange to share a
+bearer token. The bearer token is an opaque, 32-byte string that MUST be
+generated using a cryptographically secure random number generator.
+
+For requests requiring authentication, the sender includes a "PPM-Auth-Token"
+header in its HTTP request whose value is the bearer token encoded in Base64 and
+using the URL-safe alphabet defined in {{!RFC4648}}.
+
+To authenticate the request, the receiver looks up the bearer token for the
+sender as determined by the task configuration. (See {{task-configuration}}.) If
+the value of the "PPM-Auth-Token" header does not match the bearer token, then
+the receiver MUST abort with error "unauthorizedRequest".
```

Comment:
I think we should require servers to respond with HTTP 403 Forbidden here (our definition of "abort with error ..." allows servers to use any 4xx or 5xx status).

Reply:
This seems inconsistent with the guidance of BCP 56 S 8.

Reply:
Thanks for the reference, this is really useful! However I think this paragraph in [section 8](
https://datatracker.ietf.org/doc/html/rfc3205#section-8) supports using an HTTP error code:

>    A layered application should use appropriate HTTP error codes to
>      report errors resulting from information in the HTTP request-line
>      and *header fields associated with the request*.  This request
>      information is part of the HTTP protocol and errors which are
>      associated with that information should therefore be reported
>      using HTTP protocol mechanisms.

In this specific case, the error results from information in the `PPM[/DAP]-Auth-Token` HTTP header and so indicating an error in HTTP seems appropriate. However the point may be moot if we end up deciding not to dictate specific auth mechanisms at this stage or ever.

Reply:
Agreed. Good point.

Reply:
Oof, it would be useful to add an error code column to the error table. Will punt on this for now and simply add this to the text.